apps/dcos_l4lb/src/dcos_l4lb_lashup_vip_listener.erl

```diff
@@ -261,62 +289,137 @@ ip_to_integer(_IP = {A, B, C, D}) ->
     SearchStart = erlang:phash2(Name, MaxIPNum - MinIPNum),
-    add_new_name(Name, SearchStart + 1, SearchStart, State).
+    add_new_name_ipv4(Name, SearchStart + 1, SearchStart, State).
 
-add_new_name(_Name, SearchNext, SearchStart, #state{min_ip_num = MinIPNum, max_ip_num = MaxIPNum}) when
+add_new_name_ipv4(_Name, SearchNext, SearchStart, #state{min_ip_num = MinIPNum, max_ip_num = MaxIPNum}) when
     SearchNext rem (MaxIPNum - MinIPNum) == SearchStart ->
     throw(out_of_ips);
-add_new_name(Name, SearchNext, SearchStart,
-        State =
-            #state{min_ip_num = MinIPNum, max_ip_num = MaxIPNum}) ->
+add_new_name_ipv4(Name, SearchNext, SearchStart,
+        State = #state{min_ip_num = MinIPNum, max_ip_num = MaxIPNum}) ->
     ActualIPNum = MinIPNum + (SearchNext rem (MaxIPNum - MinIPNum)),
-    case ets:lookup(ip_to_name, ActualIPNum) of
+    IP = integer_to_ip(ActualIPNum),
+    case ets:lookup(ip_to_name, IP) of
+        [] ->
+            ets:insert(ip_to_name, {IP, Name}),
+            ets:insert(name_to_ip, {Name, IP});
+        _ ->
+            add_new_name_ipv4(Name, SearchNext + 1, SearchStart, State)
+    end.
+
+add_new_name_ipv6(Name, State = #state{last_ip6 = LastIP6}) ->
+    MinIP6 = dcos_l4lb_config:min_named_ip6(),
+    MaxIP6 = dcos_l4lb_config:max_named_ip6(),
+    NextIP6 = next_ip6(LastIP6, MinIP6, MaxIP6),
+    case ets:lookup(ip_to_name, NextIP6) of
         [] ->
-            ets:insert(name_to_ip, {Name, ActualIPNum}),
-            ets:insert(ip_to_name, {ActualIPNum, Name});
+            ets:insert(ip_to_name, {NextIP6, Name}),
+            ets:insert(name_to_ip, {Name, NextIP6});
         _ ->
-            add_new_name(Name, SearchNext + 1, SearchStart, State)
+            add_new_name_ipv6(Name, State#state{last_ip6 = NextIP6})
     end.
 
-ets_restart(Tab) ->
+next_ip6(IP6, MinIP6, IP6) ->
+    MinIP6;
+next_ip6({Num0, 16#ffff, 16#ffff, 16#ffff, 16#ffff, 16#ffff, 16#ffff, 16#ffff}, _, _) ->
+    {Num0 + 1, 0, 0, 0, 0, 0, 0, 0};
+next_ip6({Num0, Num1, 16#ffff, 16#ffff, 16#ffff, 16#ffff, 16#ffff, 16#ffff}, _, _) ->
+    {Num0, Num1 + 1, 0, 0, 0, 0, 0, 0};
+next_ip6({Num0, Num1, Num2, 16#ffff, 16#ffff, 16#ffff, 16#ffff, 16#ffff}, _, _) ->
+    {Num0, Num1, Num2 + 1, 0, 0, 0, 0, 0};
+next_ip6({Num0, Num1, Num2, Num3, 16#ffff, 16#ffff, 16#ffff, 16#ffff}, _, _) ->
+    {Num0, Num1, Num2, Num3 + 1, 0, 0, 0, 0};
+next_ip6({Num0, Num1, Num2, Num3, Num4, 16#ffff, 16#ffff, 16#ffff}, _, _) ->
+    {Num0, Num1, Num2, Num3, Num4 + 1, 0, 0, 0};
+next_ip6({Num0, Num1, Num2, Num3, Num4, Num5, 16#ffff, 16#ffff}, _, _) ->
+    {Num0, Num1, Num2, Num3, Num4, Num5 + 1, 0, 0};
+next_ip6({Num0, Num1, Num2, Num3, Num4, Num5, Num6, 16#ffff}, _, _) ->
+    {Num0, Num1, Num2, Num3, Num4, Num5, Num6 + 1, 0};
+next_ip6({Num0, Num1, Num2, Num3, Num4, Num5, Num6, Num7}, _, _) ->
+    {Num0, Num1, Num2, Num3, Num4, Num5, Num6, Num7 + 1}.
+
+name_to_ip(Family, Name0) ->
```

Comment:
@GoelDeepak Could you please export this function? It would be great to use in in dcos-netapi.